play-json/shared/src/main/scala/play/api/libs/json/JsConstraints.scala

```diff
@@ -71,6 +76,42 @@ trait PathReads {
       ))
     }
 
+  /**
+   * Reads a Option[T] search optional or nullable field at JsPath (field not found or null is None
+   * and other cases are Error).
+   *
+   * This method is designed for cherry-picking deeply nested fields whose path at any level may
+   * not exist.
+   *
+   * It runs through JsValue following all JsPath nodes on JsValue:
+   * - If any node in JsPath is not found => returns None
+   * - If any node in JsPath is found with value "null" => returns None
+   * - If the entire path is found => applies implicit Reads[T]
+   */
+  def deepNullable[A](path: JsPath)(implicit reads: Reads[A]): Reads[Option[A]] =
+    deepNullableWithDefault(path, None)
+
+  /**
+   * Reads an Option[T] search optional or nullable field at JsPath (field not found replaced by
+   * default value, null is None and other cases are Error).
+   *
+   * This method is designed for cherry-picking deeply nested fields whose path at any level may
+   * not exist.
+   *
+   * It runs through JsValue following all JsPath nodes on JsValue:
+   * - If any node in JsPath is not found => returns default value
+   * - If the last node in JsPath is found with value "null" => returns None
+   * - If the entire path is found => applies implicit Reads[T]
+   */
+  def deepNullableWithDefault[A](path: JsPath, defaultValue: => Option[A])(implicit reads: Reads[A]): Reads[Option[A]] =
```

Comment:
Shouldn't it been the default behaviour for the existing nullable `Reads` ?

Reply:
Yes, you are right. I'm ok to not introduce new methods and simply 'fix' the behaviour of the existing ones. 

@jroper, do you want to adapt your PR accordingly?